web/src/app/forms/FormField.js

```diff
@@ -9,206 +8,109 @@ import shrinkWorkaround from '../util/shrinkWorkaround'
-    component: p.any,
-    render: p.func,
-
-    // mapValue can be used to map a value before it's passed to the form component
-    mapValue: p.func,
-
-    // mapOnChangeValue can be used to map a changed value from the component, before it's
-    // passed to the parent form's state.
-    mapOnChangeValue: p.func,
-
-    // Adjusts props for usage with a Checkbox component.
-    checkbox: p.bool,
-
-    // Allows entering decimal number into a numeric field.
-    float: p.bool,
-
-    // fieldName specifies the field used for
-    // checking errors, change handlers, and value.
-    //
-    // If unset, it defaults to `name`.
-    name: p.string.isRequired,
-    fieldName: p.string,
-
-    // min and max values specify the range to clamp a int value
-    // expects an ISO timestamp, if string
-    min: p.oneOfType([p.number, p.string]),
-    max: p.oneOfType([p.number, p.string]),
-
-    // used if name is set,
-    // but the error name is different from graphql responses
-    errorName: p.string,
-
-    // label above form component
-    label: p.node,
-    formLabel: p.bool, // use formLabel instead of label if true
-
-    // required indicates the field may not be left blank.
-    required: p.bool,
-
-    // validate can be used to provide client-side validation of a
-    // field.
-    validate: p.func,
-
-    // a hint for the user on a form field. errors take priority
-    hint: p.node,
-
-    // disable the form helper text for errors.
-    noError: p.bool,
-
-    step: p.oneOfType([p.number, p.string]),
-  }
-
-  static defaultProps = {
-    validate: () => {},
-    mapValue: (value) => value,
-    mapOnChangeValue: (value) => value,
-  }
-
-  validate = (value) => {
+export function FormField(props) {
+  const {
+    errors,
+    value,
+    onChange,
+    addField,
+    disabled: containerDisabled,
+    optionalLabels,
+  } = useContext(FormContainerContext)
+
+  const {
+    errorName,
+    name,
+    noError,
+    component: Component,
+    render,
+    fieldName: _fieldName,
+    formLabel,
+    required,
+    validate = () => {},
+    disabled: fieldDisabled,
+    hint,
+    label: _label,
+    InputLabelProps: _inputProps,
+    mapValue = (value) => value,
+    mapOnChangeValue = (value) => value,
+    min,
+    max,
+    checkbox,
+    float,
+    ...otherFieldProps
+  } = props
+
+  const fieldName = _fieldName || name
+
+  const validateField = (value) => {
     if (
-      this.props.required &&
+      required &&
       !['boolean', 'number'].includes(typeof value) &&
       isEmpty(value)
     ) {
       return new Error('Required field.')
     }
 
-    return this.props.validate(value)
+    return validate(value)
   }
 
-  render() {
-    return (
-      <FormContainerContext.Consumer>
-        {this.renderComponent}
-      </FormContainerContext.Consumer>
-    )
+  useEffect(() => {
+    // on mount
+    const unregister = () => addField(fieldName, validateField)
+    // on unmount
+    return () => unregister
```

Comment:
`addField` should be run on mount, and it's return value ran on unmount
```suggestion
    return addField(fieldName, validateField)
```